lib/internal/Magento/Framework/Filter/Template.php

```diff
@@ -371,4 +378,59 @@ protected function getStackArgs($stack)
         }
         return $stack;
     }
+    
+    /**
+     * Filter the string as template.
+     *
+     * @param string $value
+     * @return string
+     */
+    protected function _filterLoop($value)
+    {
+        if (preg_match_all(self::LOOP_PATTERN, $value, $constructions, PREG_SET_ORDER)) {
+            foreach ($constructions as $construction) {
+
+                $full_text_to_replace = $construction[0];
+                $objectArrayData = $this->_getVariable($construction[1], '');
+                $delimiter = $construction[2];
+                $loop_text_to_replace = $construction[3];
+                
+                if (is_array($objectArrayData) || $objectArrayData instanceof Varien_Data_Collection) {
```

Comment:
You should allow more than arrays and object collections in here.
You should use `is_array($data) || ($data instanceof Traversable)`.